src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -53,6 +61,126 @@ struct CPU {
+         *  2. Four processors of four Ecores shared L2 cache
+         *
+         *  Proc ID | Socket ID | HW Core ID | Phy Core of Pcores | Logic Core of Pcores | ID of Ecore Group | Used
+         *     0         1            1                0                      1                   0             0
+         *     1         1            1                1                      0                   0             0
+         *     2         1            2                0                      2                   0             0
+         *     3         1            2                2                      0                   0             0
+         *     4         1            3                0                      0                   1             0
+         *     5         1            4                0                      0                   1             0
+         *     6         1            5                0                      0                   1             0
+         *     7         1            6                0                      0                   1             0
+         */
+        _proc = sysconf(_SC_NPROCESSORS_ONLN);
+        int _cpu_mapping[_proc][CPU_MAP_USED_PROC + 1];
+        memset(_cpu_mapping, 0, _proc * (CPU_MAP_USED_PROC + 1) * sizeof(int));
+
+        auto updateProcMapping = [&](const int nproc) {
+            if (0 == _cpu_mapping[nproc][CPU_MAP_CORE]) {
+                int core_1 = 0;
+                int core_2 = 0;
+                std::string::size_type pos = 0;
+                std::string::size_type endpos = 0;
+                std::string sub_str = "";
+
+                std::ifstream cache_1_file("/sys/devices/system/cpu/cpu" + std::to_string(nproc) +
+                                           "/cache/index0/shared_cpu_list");
+                std::ifstream cache_2_file("/sys/devices/system/cpu/cpu" + std::to_string(nproc) +
+                                           "/cache/index2/shared_cpu_list");
+                std::string cache_1_info;
+                std::getline(cache_1_file, cache_1_info);
+                std::string cache_2_info;
+                std::getline(cache_2_file, cache_2_info);
+
+                if (((endpos = cache_1_info.find(',', pos)) != std::string::npos) ||
+                    ((endpos = cache_1_info.find('-', pos)) != std::string::npos)) {
+                    sub_str = cache_1_info.substr(pos, endpos);
+                    core_1 = std::stoi(sub_str);
+                    sub_str = cache_1_info.substr(endpos + 1);
+                    core_2 = std::stoi(sub_str);
+
+                    _phy_cores++;
+                    _p_cores++;
+                    _cpu_mapping[core_1][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_2][CPU_MAP_CORE] = _phy_cores;
+                    _cpu_mapping[core_1][CPU_MAP_LOG_CORE] = _p_cores;
+                    _cpu_mapping[core_2][CPU_MAP_PHY_CORE] = _p_cores;
+
```

Comment:
Whether it is necessary to add an explanation of why the physical core is behind?

Reply:
Updated